File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/SerializableClasses/OccupancyRoomRate.vb
Imports Newtonsoft.Json
Namespace ExpediaRapid.SerializableClasses

    Public Class OccupancyRoomRate

        <JsonProperty("nightly")>
        Public Property NightlyRates As New List(Of List(Of Rate))

        <JsonProperty("stay")>
        Public Property StayRates As New List(Of Rate)

        <JsonProperty("totals")>
        Public Property OccupancyRateTotals As New Dictionary(Of String, OccupancyRateTotal)

        <JsonProperty("fees")>
        Public Property OccupancyRateFees As New Dictionary(Of String, OccupancyRateFee)

    End Class

End Namespace

File: src/libs/iVectorOne.Suppliers/ExpediaRapid/SerializableClasses/OccupancyRoomRate.cs
namespace iVectorOne.CSSuppliers.ExpediaRapid.SerializableClasses
{
    using System.Collections.Generic;
    using Newtonsoft.Json;


    public class OccupancyRoomRate
    {

        [JsonProperty("nightly")]
        public List<List<Rate>> NightlyRates { get; set; } = new List<List<Rate>>();

        [JsonProperty("stay")]
        public List<Rate> StayRates { get; set; } = new List<Rate>();

        [JsonProperty("totals")]
        public Dictionary<string, OccupancyRateTotal> OccupancyRateTotals { get; set; } = new Dictionary<string, OccupancyRateTotal>();

        [JsonProperty("fees")]
        public Dictionary<string, OccupancyRateFee> OccupancyRateFees { get; set; } = new Dictionary<string, OccupancyRateFee>();

    }

}